lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -236,7 +272,30 @@ def _snap(self, command: str, optargs: Optional[Iterable[str]] = None) -> str:
         try:
             return subprocess.check_output(_cmd, universal_newlines=True)
         except CalledProcessError as e:
-            raise SnapError("Could not %s snap [%s]: %s", _cmd, self._name, e.output)
+            raise SnapError(
+                "Snap: {!r}; command {!r} failed with output = {!r}".format(
+                    self._name, _cmd, e.output
+                )
+            )
+
+    def _snap_daemons(
+        self,
+        command: List[str],
+        services: Optional[List[str]] = None,
+    ) -> CompletedProcess:
```

Comment:
generally missing docstrings in this file

Reply:
It's not my code, only parts I added, which I doc'd. Most of it is internal methods which don't need docstrings generally.